app/models/event.rb

```diff
@@ -9,11 +9,14 @@ def start_time_must_be_lower_than_end_time
     errors.add(:start_time, "Can't be higher than end_time") if start_time.try(:>, end_time)
   end
 
+  scope :in_span, -> (starting, ending) {
+    where('start_time >= ? AND start_time <= ?', starting, ending)
```

Comment:
`where('? <= end_time AND ? >= start_time', starting, ending)` should be equivalent to your more complex condition